tests/minimal/Issue159.java

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 The JSpecify Authors
```

Comment:
@cpovirk @netdpb  Let's re-start the discussion from https://github.com/jspecify/jspecify-reference-checker/pull/165#discussion_r1503315408

I've added tests for the three fixed issues and the override test from one of the last comments. I could either remove them before we merge this or maybe move them from `minimal` to `issues` or some other sub-directory?

We can then merge this and I'll work on #164 in a new PR.

Reply:
We discussed this today and settled on moving the tests. Done in https://github.com/jspecify/jspecify-reference-checker/pull/165/commits/2588340d48ce1e94fc906d6f060c6fe069f32444 
A future cleanup can move some of them to conformance tests and keep others.